feat: add quantile aggregate function

vibhatha opened the pull request:
This PR includes the `quantile` and `approx_quantile` functions for aggregate functions.

jvanstraten commented:
From just the descriptions, implementations, and a few minutes of googling I'm not at all sure what these functions and their arguments actually do. Can you please clarify in the descriptions and include argument names?

vibhatha commented:
@jvanstraten I updated the PR.

jvanstraten reviewed (changes requested):
We discussed this function at length offline. Could you update accordingly? Right now, the description and definition just isn't good enough IMO.

vibhatha commented:
@jvanstraten I updated the PR and also for the approximate type, I created a new field called `precision` and added a description. Please check that description see if it requires modifications?

jvanstraten reviewed (approved):
~~LGTM now!~~

jvanstraten reviewed (changes requested):
Actually, correction (sorry), the same comments I gave on the other PR for precision apply here. I guess for approximate the list items should at least be monotonously increasing/decreasing if the input is sorted ascending/descending? Again, I'm also fine with just omitting its description for now since it will be a standard option.

vibhatha commented:
> Actually, correction (sorry), the same comments I gave on the other PR for precision apply here. 

I updated the description of the `precision`. 

> I guess for approximate the list items should at least be monotonously increasing/decreasing if the input is sorted ascending/descending? Again, I'm also fine with just omitting its description for now since it will be a standard option.

Shall we include options about the ordering too? If so, it shouldn't be an option of `precision` but the operator itself, right?

jvanstraten commented on extensions/functions_arithmetic.yaml:
```diff
@@ -1071,6 +1071,86 @@ aggregate_functions:
+              on saving memory bandwidth, the precision of the end result can be
+              the highest possible accuracy of an approximation. 
+
+                - EXACT: provides the highest accurate output
+                - APPROXIMATE: provides a sub-optimal output
```
> I updated the description of the `precision`.

I mean here. Or maybe you forgot to push?

vibhatha replied:
Is it okay now?

jvanstraten reviewed (changes requested):
> Shall we include options about the ordering too? If so, it shouldn't be an option of precision but the operator itself, right?

No, because aggregate function bindings already include specialized options for sorting prior to aggregation in protobuf land.

Also needs a rebase.

jvanstraten commented:
We also need to add `ordered: true` to all these implementations. I didn't realize that wasn't the default behavior. See also #312

vibhatha commented:
> We also need to add `ordered: true` to all these implementations. I didn't realize that wasn't the default behavior. See also #312

@jvanstraten updated.

jvanstraten reviewed (approved):
I think this looks good now.
